### PR TITLE
chore: remove Algebra.toSMul as instance

### PR DESCRIPTION
**Pull request by mattrobball**
For benchmarking purposes at the moment.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Algebra/Basic.lean [hunk 1]
```diff
@@ -342,7 +342,7 @@ instance (priority := 200) toModule : Module R A where
 #align algebra.to_module Algebra.toModule
 
 -- porting note: this caused deterministic timeouts later in mathlib3 but not in mathlib 4.
--- attribute [instance 0] Algebra.toSMul
+attribute [instance 0] Algebra.toSMul
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Basic.lean:345]: This does the opposite of what your PR title claims it does: it was _already_ an instance, this effectively _removes_ it as an instance.

Comment (mattrobball) [reply Mathlib/Algebra/Algebra/Basic.lean:345]: Was confused.

**Comment by mattrobball** [comment]
This does not appear to be helpful

**Comment by mattrobball** [comment]
On my machine, it built 89% of mathlib in 1:16 compared to the 1:21 to build all of master. So it does not seem like it will result in major speed improvements even if the errors are fixed.

**Pull request closed without merging**